src/App.js

```diff
@@ -28,10 +27,15 @@ class App extends Component {
     );
   }
 
-  updateAuthenticationInfo = function(authInfo) {
-    // Update the state only if authInfo is valid
-    if (authInfo && authInfo.isValid()) {
-      this.setState({ authenticationInfo: authInfo });
+  /**
+   * Update the state only if authenticationInfo is defined
+   */
+  updateAuthenticationInfo = function(authenticationInfo) {
+    if (
+      authenticationInfo &&
+      authenticationInfo instanceof AuthenticationInfo
+    ) {
+      this.setState({ authenticationInfo: authenticationInfo });
     }
   };
```

Comment:
Add a log line when you receive an authenticationInfo that you decide not to put in the state. You don't need to test it though.

Reply:
Done.